packages/ui/src/utils/TransactionUtil.ts

```diff
@@ -146,26 +147,97 @@ export const TransactionUtil = {
-          existingTransfer.quantity.numeric = quantity.toString()
+    if (transfers?.length <= 1) {
+      return transfers
+    }
+
+    // Filter out gas fee transfers (small opposite-direction amounts of same token)
+    const filteredTransfers = this.filterGasFeeTransfers(transfers)
+
+    // Merge transfers with same token and same direction
+    const mergedTransfers = filteredTransfers.reduce<TransactionTransfer[]>((acc, t) => {
+      const name = t?.fungible_info?.name
+      const existingTransfer = acc.find(
+        ({ fungible_info, direction }) =>
+          name && name === fungible_info?.name && direction === t.direction
+      )
+      if (existingTransfer) {
+        const quantity = Number(existingTransfer.quantity.numeric) + Number(t.quantity.numeric)
+        existingTransfer.quantity.numeric = quantity.toString()
+
+        existingTransfer.value = (existingTransfer.value || 0) + (t.value || 0)
+      } else {
+        acc.push(t)
+      }
+
+      return acc
+    }, [])
+
+    return mergedTransfers
+  },
+
+  filterGasFeeTransfers(transfers: TransactionTransfer[]): TransactionTransfer[] {
+    // Group transfers by token name
+    const tokenGroups = transfers.reduce<Record<string, TransactionTransfer[]>>(
+      (groups, transfer) => {
+        const tokenName = transfer?.fungible_info?.name
+        if (tokenName) {
+          if (!groups[tokenName]) {
+            groups[tokenName] = []
+          }
+          groups[tokenName].push(transfer)
+        }
+
+        return groups
+      },
+      {}
+    )
+
+    const filteredTransfers: TransactionTransfer[] = []
+
+    Object.values(tokenGroups).forEach(tokenTransfers => {
+      if (tokenTransfers.length === 1) {
+        const firstTransfer = tokenTransfers[0]
+        if (firstTransfer) {
+          filteredTransfers.push(firstTransfer)
+        }
+      } else {
+        // Multiple transfers for same token, check for gas fee pattern
+        const inTransfers = tokenTransfers.filter(t => t.direction === 'in')
+        const outTransfers = tokenTransfers.filter(t => t.direction === 'out')
+
+        if (inTransfers.length === 1 && outTransfers.length === 1) {
+          const inTransfer = inTransfers[0]
+          const outTransfer = outTransfers[0]
+
+          if (inTransfer && outTransfer) {
+            const inAmount = Number(inTransfer.quantity.numeric)
+            const outAmount = Number(outTransfer.quantity.numeric)
+
+            // If one amount is less than 10% of the other, consider it gas and filter out the gas transfer
+            if (outAmount < inAmount * GAS_FEE_THRESHOLD) {
+              filteredTransfers.push(inTransfer)
+            } else if (inAmount < outAmount * GAS_FEE_THRESHOLD) {
+              filteredTransfers.push(outTransfer)
+            } else {
+              filteredTransfers.push(...tokenTransfers)
+            }
+          } else {
+            filteredTransfers.push(...tokenTransfers)
```

Comment:
isnt there and edge case here where you are transferring small amounts? how do we distinguish?